test/test_core_version.py

```diff
@@ -0,0 +1,31 @@
+from unittest import TestCase
+
+from voicevox_engine.utility.core_version import (
+    get_latest_core_version,
+    parse_core_version,
+)
+
+
+class TestCoreVersion(TestCase):
+    def test_parse_core_version(self):
+        parse_core_version("0.1.0")
+        parse_core_version("0.10.preview.1")
+        parse_core_version("0.10.0")
+        parse_core_version("0.14.0")
+        parse_core_version("0.14.0.preview.1")
+        parse_core_version("0.14.0.preview.10")
```

Comment:
VOICEVOXで取り入れてる（ドキュメント化はまだですが 🙇 ）プレビュー版でのバージョン表記に合わせてみました！
```suggestion
        parse_core_version("0.1.0")
        parse_core_version("0.10-preview.1")
        parse_core_version("0.10.0")
        parse_core_version("0.14.0")
        parse_core_version("0.14.0-preview.1")
        parse_core_version("0.14.0-preview.10")
```